domain/rates/rates_service.go

```diff
@@ -0,0 +1,88 @@
+package rates
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"sync"
+	"time"
+
+	"github.com/bitcoin-sv/spv-wallet-web-backend/config"
+	"github.com/rs/zerolog"
+	"github.com/spf13/viper"
+)
+
+// RatesService is a service for fetching and caching BSV exchange rates.
+type RatesService struct {
+	exchangeRate *ExchangeRate
+
+	mutex     sync.Mutex
+	lastFetch time.Time
+}
+
+// ExchangeRate is a struct that contains exchange rate data.
+type ExchangeRate struct {
+	Rate float64
+}
+
+func NewRatesService(log *zerolog.Logger) *RatesService {
+	s := &RatesService{
+		exchangeRate: nil,
+	}
+
+	err := s.makeExchangeRate()
+	if err != nil {
+		log.Error().Msg(err.Error())
+	}
+
+	return s
+}
+
+// GetExchangeRate returns the current exchange rate.
+func (s *RatesService) GetExchangeRate() (*ExchangeRate, error) {
+	err := s.makeExchangeRate()
+	if err != nil {
+		return nil, err
+	}
+
+	return s.exchangeRate, nil
+}
+
+func (s *RatesService) makeExchangeRate() error {
```

Comment:
Maybe a slighty better name:
```suggestion
func (s *RatesService) loadExchangeRate() error {
```

Reply:
Done